src/libopensc/pkcs15-pubkey.c

```diff
@@ -680,57 +675,68 @@ sc_pkcs15_encode_pubkey_ec(sc_context_t *ctx, struct sc_pkcs15_pubkey_ec *key,
 		u8 **buf, size_t *buflen)
 {
 	struct sc_asn1_entry asn1_ec_pointQ[C_ASN1_EC_POINTQ_SIZE];
+	size_t  key_len;
+	volatile int gdb_test = 0; /* so can reset via gdb for testing new  way */
 
 	LOG_FUNC_CALLED(ctx);
 	sc_copy_asn1_entry(c_asn1_ec_pointQ, asn1_ec_pointQ);
-	sc_format_asn1_entry(asn1_ec_pointQ + 0, key->ecpointQ.value, &key->ecpointQ.len, 1);
+
+	if (gdb_test == 1) {
+		key_len  = key->ecpointQ.len * 8; /* encode in bit string */
+		sc_format_asn1_entry(asn1_ec_pointQ + 0, key->ecpointQ.value, &key_len, 1);
+	} else {
+		key_len  = key->ecpointQ.len;
+		sc_format_asn1_entry(asn1_ec_pointQ + 1, key->ecpointQ.value, &key_len, 1);
+	}
```

Comment:
before merging we will certainly need to remove the debugging variables like this.

Reply:
This is the outstanding question of #3000 Did OpenSC return the CKA_EC_POINT in the wrong ASN.1?
It also looks like other packages followed our lead.  This would not be easy to fix. The remedy is to use the  CKA_PUBKEY_KEY_INFO as SPKI introduced in KCS11 2.40. 

Other commits in this PR adds CKA_PUBKEY_KEY_INFO to OpenSC  and will accept CKA_EC_POINT in either BIT STRING or OCTET STRING.

Reply:
I would say that given that we believe the right format should be the bit string, we should output bit string and properly test it with other applications that might expect the other way round and fix them, ideally early in the release cycle. Having this in the code comment should be quite enough. Other option might be to provide some environment variable or configuration option for backward compatibility to satisfy unfixed applications.

Reply:
I am afraid that OpenSC was one of the first smartcard pkcs11 project to include ECDSA and got it wrong and other projects picked up on it.

You at RedHat have contacts with p11-kit and PKCS11 OASIS people. Can you ask them if we got to wrong, and what it would to convert?  Or is the solution for every one to use CKA_PUBLIC_KEY_INFO that returns a SPKI format.

Reply:
I had some discussion on the OASIS PKCS#11 mailing list to get the format clarified, but as far as I remember, they were not able to provide much more clarity (I do not follow the discussions there much now)

There was recently exactly this topic discussed here:

https://lists.oasis-open.org/archives/pkcs11/202310/msg00008.html

There seems to be some changes in the PKCS#11 3.1.

Let me know if there is something else to bring up there.

Reply:
I removed the `if (gdb_test == 1) {` and related code, as I closed #3000. The code returns the EC_Point as OCTET STRING as it did before.  When returned as part of a SPKI, its in a BIT STRING produced by some other code.